frontend.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+
+/*
+ * Outputs the HTML required to display the text in the about me section.
+ *
+ * @param PDO $db The database object to retrieve the about me information from.
+ *
+ * @return string Returns the Name, Title and Description wrapped in the correct html tags.
+ */
+function displayAboutMe(PDO $db) {
+    $stmt = $db->prepare('SELECT `name`,`title`,`desc` FROM `about`');
+    $stmt->execute();
+    $result = $stmt->fetch();
+    $name = '<h1>' . $result['name'] . '</h1>';
+    $title = '<h2>' . $result['title'] . '</h2>';
+    $desc = '<p>' . $result['desc'] . '</p>';
+    return $name . $title . $desc;
+}
+
+/*
+ * Outputs the HTML required to display the badges in the about me section.
+ *
+ * @param PDO $db The database object to retrieve the about me information from.
+ *
+ * @return string Returns each badge in the database, wrapped in an <i> tag.
+ */
+function displayBadges(PDO $db) {
+    $stmt = $db->prepare('SELECT `value` FROM `badges`');
+    $stmt->execute();
+    $entries = $stmt->fetchAll();
+    $output = '';
+    foreach ($entries as $entry) {
+        $output .= '<i class="' . $entry['value'] . '"></i>';
+    }
+    return $output;
+}
+
+/*
+ * Clamps the project ID to an existing project in the database. If the ID is higher than any existing number,
+ * we loop back around to the first project ID. If it is lower, we loop back around to the last project ID. If
+ * in the event the autoincrement screws up and we have missing numbers we will round to the nearest existing number.
+ *
+ * @param int The target project ID recieved from the GET string.
+ *
+ * @return int The corrected ID of the project we will be displaying.
+ */
+function clampProjectID(PDO $db, int $id, bool $up) {
+    $stmt = $db->prepare('SELECT `id` FROM `projects`');
+    $stmt->execute();
+    $select = $stmt->fetchAll();
+    $array = [];
+    if(in_array($id, $array)) {
+        return $id;
+    }
+    foreach ($select as $entry) {
+        array_push($array, (int)$entry['id']);
+    }
+    if($id < min($array)) {
+        return max($array);
+    } else if($id > max($array)) {
+        return min($array);
+    }
+    if($up) {
+        roundValueUp($array, $id);
+    }
+    return roundValueDown($array, $id);
+}
+
+/*
+ * This function rounds the given value upward to the nearest existing value in the array.
+ *
+ * @param array $array The array to search.
+ *
+ * @param numeric $val The value to find the closest match for.
```

Comment:
params should be grouped together, no line breaks between them

Reply:
fixed in 5d4ede8